lib/binary_parser.ts

```diff
@@ -1161,11 +1414,87 @@ export class Parser {
     }
   }
 
+  private generate_encodeBit(ctx: Context) {
+    // TODO find better method to handle nested bit fields
+    const parser = JSON.parse(JSON.stringify(this));
+    parser.varName = ctx.generateVariable(parser.varName);
+    ctx.bitFields.push(parser);
+
+    if (
+      !this.next ||
+      (this.next && ["bit", "nest"].indexOf(this.next.type) < 0)
+    ) {
+      let sum = 0;
+      ctx.bitFields.forEach((parser) => {
+        sum += parser.options.length as number;
+      });
+
+      if (sum <= 8) {
+        sum = 8;
+      } else if (sum <= 16) {
+        sum = 16;
+      } else if (sum <= 24) {
+        sum = 24;
+      } else if (sum <= 32) {
+        sum = 32;
+      } else {
+        throw new Error(
+          "Currently, bit field sequences longer than 4-bytes is not supported.",
+        );
+      }
+
+      const isBitLittleEndian =
+        this.endian === "le" && this.encoderOpts.bitEndianess;
+      const tmpVal = ctx.generateTmpVariable();
+      const boundVal = ctx.generateTmpVariable();
+      ctx.pushCode(`var ${tmpVal} = 0;`);
+      ctx.pushCode(`var ${boundVal} = 0;`);
+      let bitOffset = 0;
+      ctx.bitFields.forEach((parser) => {
+        ctx.pushCode(
+          `${boundVal} = (${parser.varName} & ${
+            (1 << (parser.options.length as number)) - 1
+          });`,
+        );
+        ctx.pushCode(
+          `${tmpVal} |= (${boundVal} << ${
+            isBitLittleEndian
+              ? bitOffset
+              : sum - (parser.options.length as number) - bitOffset
+          });`,
+        );
+        ctx.pushCode(`${tmpVal} = ${tmpVal} >>> 0;`);
+        bitOffset += parser.options.length as number;
+      });
+      if (sum == 8) {
+        ctx.pushCode(`smartBuffer.writeUInt8(${tmpVal});`);
+      } else if (sum == 16) {
+        ctx.pushCode(`smartBuffer.writeUInt16BE(${tmpVal});`);
+      } else if (sum == 24) {
+        const val1 = ctx.generateTmpVariable();
+        const val2 = ctx.generateTmpVariable();
+        ctx.pushCode(`var ${val1} = (${tmpVal} >>> 8);`);
+        ctx.pushCode(`var ${val2} = (${tmpVal} & 0x0ff);`);
+        ctx.pushCode(`smartBuffer.writeUInt16BE(${val1});`);
+        ctx.pushCode(`smartBuffer.writeUInt8(${val2});`);
+      } else if (sum == 32) {
+        ctx.pushCode(`smartBuffer.writeUInt32BE(${tmpVal});`);
+      }
+
+      ctx.bitFields = [];
+    }
+  }
+
   private generateSeek(ctx: Context) {
     const length = ctx.generateOption(this.options.length!);
     ctx.pushCode(`offset += ${length};`);
   }
 
+  private generate_encodeSeek(ctx: Context) {
+    const length = ctx.generateOption(this.options.length!);
+    ctx.pushCode(`smartBuffer.writeBuffer(Buffer.alloc(${length}));`);
+  }
```

Comment:
This seems to not be compatible with parsers which use negative `seek` values, which is a supported feature https://github.com/keichi/binary-parser#seekreloffset